lib/rust/lexer/definition/src/token.rs

```diff
@@ -80,6 +85,14 @@ impl Token {
         Token{shape,length,offset}
     }
 
+    /// Construct a token representing
+    pub fn Annotation(name_str:impl Str, offset:usize) -> Token {
```

Comment:
please put all these "smart constructors" in a separate `impl Token` block with a separate section